subprojects/testfx-core/src/main/java/org/testfx/util/DebugUtils.java

```diff
@@ -0,0 +1,213 @@
+/*
+ * Copyright 2013-2014 SmartBear Software
+ * Copyright 2014-2017 The TestFX Contributors
+ *
+ * Licensed under the EUPL, Version 1.1 or - as soon they will be approved by the
+ * European Commission - subsequent versions of the EUPL (the "Licence"); You may
+ * not use this work except in compliance with the Licence.
+ *
+ * You may obtain a copy of the Licence at:
+ * http://ec.europa.eu/idabc/eupl.html
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the Licence is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied. See the Licence for the
+ * specific language governing permissions and limitations under the Licence.
+ */
+package org.testfx.util;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Set;
+import java.util.function.Function;
+
+import javafx.event.Event;
+import javafx.scene.Node;
+import javafx.scene.input.KeyCode;
+import javafx.scene.input.MouseButton;
+
+import org.hamcrest.Matcher;
+import org.testfx.api.FxRobot;
+import org.testfx.api.FxToolkit;
+import org.testfx.service.support.FiredEvents;
+
+/**
+ * Utility class for displaying additional info or running code when a test fails using
+ * {@link org.testfx.api.FxAssert#verifyThat(Node, Matcher)} or its related methods.
+ *
+ * <p>
+ *     Quickly chain things together using {@link #compose(Function[])} or use standard handlers like
+ *     {@link #informedErrorMessage(String, boolean, FxRobot, boolean, boolean)}. Any {@code tab} parameters in methods
+ *     are there to specify what to insert as a tab. This could be a tab or a number of spaces. Default tab
+ *     value is three spaces (e.g. {@code "   "}).
+ * </p>
+ */
+public final class DebugUtils {
```

Comment:
Should we maybe say "are there to specify what text is inserted for spacing purposes. This defaults to blah blah". What I mean is instead of explicitly calling it "tab", maybe rename to `spacing`?